src/trio/_abc.py

```diff
@@ -698,3 +704,53 @@ class Channel(SendChannel[T], ReceiveChannel[T]):
     `ReceiveChannel` interfaces, so you can both send and receive objects.
 
     """
+
+
+class TaskSpawner(Protocol):
```

Comment:
SupportsStartSoon is the standard

Reply:
SupportsStartsSoon is a stupid name

Reply:
Yeah the `SupportsWhatever` names are mostly because ATM we use protocols for self-types, so we can be like `def f(self: Something[SupportsSomething])` and it reads nicer (we want the generic argument to support `.something`). IDK if that's a good naming idea for protocols that aren't used like that.

Reply:
Alternative idea: `NurseryLike` (the "task spawner" in the `start_soon` docstring is a bit awkward, though it's workable so `TaskSpawner` is... fine.)

Reply:
`SupportsClose` is the first example of a protocol in the PEP that defined protocols. https://peps.python.org/pep-0544/#defining-a-protocol

The `typing` module defines a bunch more using the same naming scheme: https://docs.python.org/3/library/typing.html#protocols

I think being consistent/recognizable is valuable here. I think `SupportsStartSoon` and `NurseryLike` are both fine. I disprefer `TaskSpawner` (a nursery is arguably the *target* of task spawning, not the agent of it) but you've made your position abundantly clear so <shrug>.